tools/copyblocks/main.go

```diff
@@ -171,13 +187,72 @@ func copyBlocks(ctx context.Context, cfg config, logger log.Logger, m *metrics)
 		disabledUsers[u] = struct{}{}
 	}
 
-	client, err := storage.NewClient(ctx)
-	if err != nil {
-		return errors.Wrapf(err, "failed to create client")
-	}
+	var sourceBucket, destBucket bucket
+	switch strings.ToLower(cfg.service) {
+	case serviceGCS:
+		client, err := storage.NewClient(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create client")
+		}
+		sourceBucket = &gcsBucket{
+			BucketHandle: *client.Bucket(cfg.sourceBucket),
+			name:         cfg.sourceBucket,
+		}
+		destBucket = &gcsBucket{
+			BucketHandle: *client.Bucket(cfg.destBucket),
+			name:         cfg.destBucket,
+		}
+	case serviceABS:
+		if cfg.azureSourceAccountKey == "" || cfg.azureSourceAccountName == "" {
+			return errors.New("the azure source bucket's account name and account key are required")
+		}
+		if cfg.azureDestinationAccountKey == "" || cfg.azureDestinationAccountName == "" {
+			return errors.New("the azure destination bucket's account name and account key are required")
+		}
```

Comment:
[nit] We could include the CLI flag names here (eg. `--azure-source-account-name`) to make the issue even clearer